### ISSUE TITLE
Bubbling is ignored when one of handlers in the chain fails

### ISSUE THREAD

**Issue by WoZ** [Seldaek/monolog#1673]
Monolog version 1, 2, 3

The default behavior is to ignore bubbling if something goes wrong inside a handler.

Look at this part of the code.

https://github.com/Seldaek/monolog/blob/60ad5183b5e5d6c9d4047e9f3072d36071dcc161/src/Monolog/Logger.php#L325-L334

Imo, errors in some handlers must not affect other handlers if `bubble` options is set.

**Comment by Seldaek** [comment]
It is the default mode that errors abort operation so you notice any logging issue hopefully. If you don't care or have unreliable handlers you can wrap them in WhatFailureGroupHandler instance(s) to suppress the errors and keep going.

**Comment by WoZ** [comment]
@Seldaek Thank you for reply. All http/tcp-based handlers are unreliable. Socket stream does the same.

https://github.com/Seldaek/monolog/blob/60ad5183b5e5d6c9d4047e9f3072d36071dcc161/src/Monolog/Handler/SocketHandler.php#L376-L403

Anyway, maybe it's a good idea to explain it in docs? Because, this topic is confusing and users expect imho the behaviour I described.

P.S. For my purposes I implemented also error handling of handle function fails. I also log that data to get info when tcp-based handler fails. You may check this approach for the inspiration. https://github.com/BETER-CO/yii2-beter-logging/blob/master/src/Handler/ProcessHandleResultTrait.php

**Comment by WoZ** [comment]
So, I added 2 features: stats for handling time distribution and the amount of failures/successful calls. If you are ok, I'm ready to bring those handlers to monolog as HandlerWithStats and some extended features to track handle exceptions and process them from the application.

**Comment by Seldaek** [comment]
Yes, socket handlers are unreliable, and I wouldn't recommend using them for that reason. Using UDP or syslog then some background process pushing logs to another system via TCP in a more reliable manner and out of band from the application process makes a lot more sense IMO. But for small scale stuff the socket handlers are ok and convenient as things work most of the time.

I am not really interested in adding more handlers to maintain here so if you like to publish yours please do so as a standalone package - you can then list it here https://github.com/Seldaek/monolog/wiki/Third-Party-Packages
